test/e2e/scheduling/preemption.go

```diff
@@ -472,6 +468,33 @@ func createPauseRS(f *framework.Framework, conf pauseRSConfig) *appsv1.ReplicaSe
 
 func runPauseRS(f *framework.Framework, conf pauseRSConfig) *appsv1.ReplicaSet {
 	rs := createPauseRS(f, conf)
-	framework.ExpectNoError(replicaset.WaitForReplicaSetTargetAvailableReplicas(f.ClientSet, rs, conf.Replicas))
+	framework.ExpectNoError(replicaset.WaitForReplicaSetTargetAvailableReplicasWithTimeout(f.ClientSet, rs, conf.Replicas, framework.PodGetTimeout))
 	return rs
 }
+
+func createPod(f *framework.Framework, conf pausePodConfig) *v1.Pod {
+	namespace := conf.Namespace
+	if len(namespace) == 0 {
+		namespace = f.Namespace.Name
+	}
+	pod, err := f.ClientSet.CoreV1().Pods(namespace).Create(initPausePod(f, conf))
+	framework.ExpectNoError(err)
+	return pod
+}
+
+// waitForPreemptingWithTimeout verifies if 'pod' is preempting within 'timeout', specifically it checks:
+// - if the 'status.NominatedNodeName' field of preemptor 'pod' has been set
```

Comment:
if this is being used to block before we do the check on the preempted pods, we need to wait for spec.nodeName, see https://github.com/kubernetes/kubernetes/pull/86400/files#r359691588

Reply:
done.

Reply:
this still returns as soon as `len(pod.Status.NominatedNodeName) > 0`, which is too soon. we need to wait for the preempting pod to actually get scheduled (`len(pod.Spec.NodeName) > 0`)